bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/handler/DoorbellHandler.java

```diff
@@ -108,35 +99,32 @@ public DoorbirdHandler(Thing thing, TimeZoneProvider timeZoneProvider, HttpClien
 
     @Override
     public void initialize() {
-        updateStatus(ThingStatus.UNKNOWN);
-
-        config = getConfigAs(DoorbirdConfiguration.class);
-        if (StringUtils.isEmpty(config.doorbirdId)) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Doorbird id not provided");
-            return;
-        }
-        if (StringUtils.isEmpty(config.doorbirdHost)) {
+        config = getConfigAs(DoorbellConfiguration.class);
+        String host = config.doorbirdHost;
+        if (host == null) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Doorbird host not provided");
             return;
         }
-        if (StringUtils.isEmpty(config.userId)) {
+        String user = config.userId;
+        if (user == null) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "User ID not provided");
             return;
         }
-        if (StringUtils.isEmpty(config.userPassword)) {
+        String password = config.userPassword;
+        if (password == null) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "User password not provided");
             return;
         }
-        // Create the basic authorization string for the HTTP requests
-        authorization = new String(Base64.getEncoder().encode((config.userId + ":" + config.userPassword).getBytes()),
-                StandardCharsets.UTF_8);
-
+        api.setAuthorization(host, user, password);
+        api.setHttpClient(httpClient);
         startImageRefreshJob();
         startUDPListenerJob();
+        updateStatus(ThingStatus.ONLINE);
     }
 
     @Override
     public void dispose() {
+        super.dispose();
```

Comment:
It is better practice to call `super.dispose()` last.

Reply:
Yep